contracts/RewardPool.vy

```diff
@@ -222,50 +207,18 @@ def _claim(addr: address, last_token_time: uint256) -> uint256:
         return 0
 
     week_cursor: uint256 = self.time_cursor_of[addr]
-    if week_cursor == 0:
-        # Need to do the initial binary search
-        user_epoch = self._find_timestamp_user_epoch(addr, _start_time, max_user_epoch)
-    else:
-        user_epoch = self.user_epoch_of[addr]
-
-    if user_epoch == 0:
-        user_epoch = 1
-
-    user_point: Point = VEYFI.point_history(addr, user_epoch)
-
-    if week_cursor == 0:
-        week_cursor = (user_point.ts + WEEK - 1) / WEEK * WEEK
-
     if week_cursor >= last_token_time:
         return 0
 
-    if week_cursor < _start_time:
-        week_cursor = _start_time
-    old_user_point: Point = empty(Point)
-
     # Iterate over weeks
     for i in range(50):
         if week_cursor >= last_token_time:
             break
-
-        if week_cursor >= user_point.ts and user_epoch <= max_user_epoch:
-            user_epoch += 1
-            old_user_point = user_point
-            if user_epoch > max_user_epoch:
-                user_point = empty(Point)
-            else:
-                user_point = VEYFI.point_history(addr, user_epoch)
-
-        else:
-            # Calc
-            # + i * 2 is for rounding errors
-            dt: int128 = convert(week_cursor - old_user_point.ts, int128)
-            zero: int128 = 0
-            balance_of: uint256 = convert(max(old_user_point.bias - dt * old_user_point.slope, zero), uint256)
-            if balance_of == 0 and user_epoch > max_user_epoch:
+        balance_of: uint256 = VEYFI.balanceOf(addr, week_cursor)
+        if balance_of == 0 and user_epoch > max_user_epoch:
```

Comment:
observations:
- init of `user_epoch` has been removed in this commit so this check seems redundant.
- at the end of `#_claim` the user_epoch is set to `-1` and pushed into the `Claimed` event.